wc-src/vf-voice.js

```diff
@@ -0,0 +1,83 @@
+import Vex from '../src/index.js';
+
+const template = document.createElement('template');
+template.innerHTML = `
+`;
+
+function concat(a, b) {
+  return a.concat(b);
+}
+
+export class VFVoice extends HTMLElement {
+  constructor() {
+    super();
+
+    this.attachShadow({ mode:'open' });
+    this.shadowRoot.appendChild(document.importNode(template.content, true));
+
+    // Defaults
+    this.stem = 'up';
+    this.autoBeam = false;
+    this.notes = [];
+    this.beams = [];
+
+    this._vf = undefined;
+    this._score = undefined;
+  }
+
+  connectedCallback() {
+    this.stem = this.getAttribute('stem') || this.stem;
+    this.autoBeam = this.hasAttribute('autoBeam');
+    this.notesText = this.textContent.trim();
+
+    const getFactoryEvent = new CustomEvent('getFactory', { bubbles: true });
+    this.dispatchEvent(getFactoryEvent);
+
+    const getScoreEvent = new CustomEvent('getScore', { bubbles: true });
+    this.dispatchEvent(getScoreEvent);
+  }
+
+  set vf(value) {
+    this._vf = value;
+    this.createNotes();
+  }
+
+  set score(value) {
+    this._score = value;
+    this.createNotes();
+  }
+
+  createNotes() {
+    if (this._vf && this._score) {
+      const notes = this.createNotesFromText();
+      this.notes.push(notes);
+      if (this.autoBeam) {
+        this.beams.push(this.autoGenerateBeams(notes));
+      }
+
+      this.notes = this.notes.reduce(concat);
```

Comment:
An anonymous function may be cleaner here, to get rid of the top level function:
```js
this.notes.reduce((a, b) => a.concat(b));
```
But how deep is the result of createNotesFromText? Does it create a 1-d array, 2-d array? Curious

Reply:
createNotesFromText gives a 1-d array

Reply:
Then I _think_ we can replace this
```
const notes = this.createNotesFromText();
this.notes.push(notes);
this.notes = this.notes.reduce(concat);
```
with
```
this.notes = this.createNotesFromText();
```
What do you think?

Reply:
This kind of has to do with going forward, when the vf-tuplet and vf-beam components get added. Since those have their own notes, I was pushing them all to `this.notes`, and then reducing the array once everything is pushed

Reply:
Oh like in a future commit?

Then I would suggest either using [spread syntax](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax) to do `this.notes.push(...notes)`, or use [flat](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flat) `this.notes = this.notes.flat()`. Then we can get rid of the top level concat method.